src/app/components/Tag/Tag.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react'
+import type { ReactNode } from 'react'
+import styled from 'styled-components'
+
+type TagProps = {
+  children: ReactNode
+  active: boolean
+}
+export default function Tag({ children }: TagProps): JSX.Element {
+  const [btnClicked, setBtnClicked] = useState<boolean>(false)
+
+  function handleClick() {
+    if (children === 'Hotel') {
+      localStorage.setItem('ActiveSearchTag', '19014')
+      setBtnClicked(!btnClicked)
```

Comment:
Could you put this before the if loop? If this works, you could save some lines of code 🤔